src/main/java/edu/greenblitz/robotName/subsystems/swerve/Modules/mk4iSwerveModule/MK4iSwerveConstants.java

```diff
@@ -1,41 +1,102 @@
-	public static final double ANGLE_TICKS_TO_WHEEL_TO_RPM = RobotConstants.General.Motors.FALCON_VELOCITY_UNITS_PER_RPM / ANGULAR_GEAR_RATIO;
-	public static final double LINEAR_TICKS_TO_METERS_PER_SECOND = RobotConstants.General.Motors.FALCON_VELOCITY_UNITS_PER_RPM / LINEAR_GEAR_RATIO * WHEEL_CIRCUMFERENCE / 60;
-	public static final double ANGLE_TICKS_TO_RADIANS = RobotConstants.General.Motors.FALCON_TICKS_PER_RADIAN / ANGULAR_GEAR_RATIO;
-	public static final double MAG_ENCODER_TICKS_TO_FALCON_TICKS = 2 * Math.PI / ANGLE_TICKS_TO_RADIANS;
-	
-	public static final PIDObject ANGULAR_PID = new PIDObject().withKp(0.05).withMaxPower(1.0).withFF(0);
-	public static final GBFalcon.FalconConfObject ANGULAR_FALCON_CONFIG_OBJECT = new GBFalcon.FalconConfObject().withNeutralMode(NeutralMode.Brake).withCurrentLimit(30).withRampRate(RobotConstants.General.RAMP_RATE_VAL).withInverted(true).withPID(ANGULAR_PID);
 	
-	public static final PIDObject LINEAR_PID = new PIDObject().withKp(0.0003).withMaxPower(0.5);
-	public static final GBFalcon.FalconConfObject LINEAR_FALCON_CONF_OBJECT = new GBFalcon.FalconConfObject()
-			.withNeutralMode(NeutralMode.Brake)
-			.withCurrentLimit(40)
-			.withRampRate(RobotConstants.General.RAMP_RATE_VAL)
-			.withPID(LINEAR_PID);
+	public static final double WHEEL_RADIUS = Units.inchesToMeters(2);
+	public static final double WHEEL_CIRCUMFERENCE = WHEEL_RADIUS * 2 * Math.PI;
+	public static final double LINEAR_REVOLUTIONS_TO_METERS = RobotConstants.General.Motors.FALCON_REVOLUTIONS_PER_RADIAN * WHEEL_CIRCUMFERENCE / 2 / Math.PI / LINEAR_GEAR_RATIO;
+	public static final double ANGLE_REVOLUTIONS_TO_WHEEL_TO_RPS = 1 / ANGULAR_GEAR_RATIO;
+	public static final double LINEAR_REVOLUTIONS_TO_METERS_PER_SECOND = 1 / LINEAR_GEAR_RATIO * WHEEL_CIRCUMFERENCE;
+	public static final double ANGLE_REVOLUTIONS_TO_RADIANS = RobotConstants.General.Motors.FALCON_REVOLUTIONS_PER_RADIAN / ANGULAR_GEAR_RATIO;
+	public static final TalonFXConfiguration ANGULAR_FALCON_CONFIG_OBJECT = new TalonFXConfiguration();
+	static {
+		Slot0Configs PID_CONFIGS = new Slot0Configs();
+		PID_CONFIGS.kP = 1;
+		PID_CONFIGS.kI = 0;
+		PID_CONFIGS.kD = 0;
+		PID_CONFIGS.kS = 0;
+		PID_CONFIGS.kV = 0;
+
+		CurrentLimitsConfigs CURRENT_LIMITS_CONFIGS = new CurrentLimitsConfigs();
+		CURRENT_LIMITS_CONFIGS.StatorCurrentLimitEnable = true;
+		CURRENT_LIMITS_CONFIGS.StatorCurrentLimit = 40;
+		CURRENT_LIMITS_CONFIGS.SupplyCurrentThreshold = 2;
+
+		ClosedLoopRampsConfigs CLOSED_LOOP_RAMPS_CONFIGS = new ClosedLoopRampsConfigs();
+		CLOSED_LOOP_RAMPS_CONFIGS.VoltageClosedLoopRampPeriod = 0.1;
+
+        var MOTOR_OUTPUT_CONFIGS = new MotorOutputConfigs();
+		MOTOR_OUTPUT_CONFIGS.NeutralMode = NeutralModeValue.Brake;
+		MOTOR_OUTPUT_CONFIGS.Inverted = InvertedValue.CounterClockwise_Positive; //true
+
+		MotionMagicConfigs MOTION_MAGIC_CONFIGS = new MotionMagicConfigs();
+		MOTION_MAGIC_CONFIGS.MotionMagicAcceleration = 10;
```

Comment:
R these accurate or needs to be calibrated on new robot?

Reply:
It needs to be calibrated on the robot

![image](https://github.com/GreenBlitz/CRRRRESCENDO/assets/151079574/c9fdea91-92cc-4c66-92d0-35599d4973ab)